Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -31,6 +31,35 @@ function Format-Splat {
     "$parameters" # String join the array.
 }
 
+function Generate-TlsError {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)] $exception 
+    )
+
+    if ($exception -eq $null)
+    {
+        return
+    }
+
+    $isWebException = $true
+    $innerException = $exception
+    while ($innerException.GetType() -ne [System.Net.WebException])
+    {
+        $innerException = $innerException.InnerException
+        if ($innerException -eq $null)
+        {
+            $isWebException = $false
+            break
+        }
+    }
+
+    if (($isWebException -eq $true) -and ($innerException.InnerException -ne $null) -and ($innerException.InnerException.GetType() -eq [System.IO.IOException]))
```

Comment:
what happens in else case? don't we need to error out task there as well?

Reply:
Else means that no TLS error is there. So we don't do anything. Original exception handling is not changed